### ISSUE TITLE
the loss of unsupervised training could not decline

### ISSUE THREAD

**Issue by lovejing0306** [facebookresearch/deepcluster#31]
I trained mnist dataset used "unsupervised training", but the loss could not decline.
this is the training process:
###### Epoch [4] ######
Time: 8.809 s
Clustering loss: 953.089
ConvNet loss: 2.084
NMI against previous assignment: 0.227
#######################

Compute features
0 / 32	Time: 0.227 (0.227)
k-means loss evolution: [1795.6045   978.0879   971.03766  959.80444  957.0581   956.3157
  955.26294  954.6638   954.14594  953.74384  953.33545  952.984
  952.7616   952.4509   952.3556   952.30585  952.2396   952.1643
  952.1643   952.1643 ]
k-means time: 1 s
Epoch: [5][0/32]	Time: 0.139 (0.139)	Data: 0.129 (0.129)	Loss: 2.3899 (2.3899)
###### Epoch [5] ######
Time: 8.807 s
Clustering loss: 952.164
ConvNet loss: 2.133
NMI against previous assignment: 0.243
#######################

###### Epoch [264] ######
Time: 8.869 s
Clustering loss: 950.949
ConvNet loss: 1.832
NMI against previous assignment: 0.336
#######################

Compute features
0 / 32	Time: 0.233 (0.233)
k-means loss evolution: [1795.1869   977.5625   972.47687  967.0636   960.8979   955.13916
  952.04254  949.78107  945.9483   944.14136  943.93304  943.7497
  943.62286  943.5072   943.5072   943.5072   943.5072   943.5072
  943.5072   943.5072 ]
k-means time: 1 s
Epoch: [265][0/32]	Time: 0.140 (0.140)	Data: 0.129 (0.129)	Loss: 2.5108 (2.5108)
###### Epoch [265] ######
Time: 8.867 s
Clustering loss: 943.507
ConvNet loss: 1.752
NMI against previous assignment: 0.251
#######################

I use different learning rates lr = 0.001, 0.005, 0.01, 0.05, but the loss never decline.
Can you help me? 
thanks

**Comment by mathildecaron31** [comment]
Hi,
The classification loss you're observing is the one at the first iteration of the epoch ([0/32]). The final prediction layer (`top_layer`) is reinitialized at the beginning of each epoch so it makes sense that at the very beginning of the epoch (ie first iteration) the loss is high.
You should look the loss at the end of the epoch.
Hope that helps

**Comment by lovejing0306** [comment]
> Hi,
> The classification loss you're observing is the one at the first iteration of the epoch ([0/32]). The final prediction layer (`top_layer`) is reinitialized at the beginning of each epoch so it makes sense that at the very beginning of the epoch (ie first iteration) the loss is high.
> You should look the loss at the end of the epoch (which actually decreases from 2.133 to 1.752 in your example).
> Hope that helps

thanks
